### PR TITLE
GH-1555 Improve FedX Demo code and examples

### PR DESCRIPTION
**Pull request by aschwarte10**
GitHub issue resolved: #1555  

Improve the examples and demo snippets

### CODE REVIEW

File: tools/federation/src/test/java/demos/MonitorRequestsDemo.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			try (TupleQueryResult res = query.evaluate()) {
+
+				int count = 0;
+				while (res.hasNext()) {
+					res.next();
+					count++;
+				}
+
+				System.out.println("# Done, " + count + " results");
+			}
```
Comment (hmottestad) [inline tools/federation/src/test/java/demos/MonitorRequestsDemo.java]: Btw. Counting is nicer with 
```
try(.... stream = Iterations.stream(query.evaluate)){
   .... count = stream.count();
}
```

Comment (jeenbroekstra) [reply tools/federation/src/test/java/demos/MonitorRequestsDemo.java]: Or, even simpler: 

```
 List<BindingSet> result = QueryResults.asList(query.evaluate());
 System.out.println("# Done, " + result.size() + " results");
```

But it depends on what you're trying to illustrate of course.

Comment (hmottestad) [reply tools/federation/src/test/java/demos/MonitorRequestsDemo.java]: That keeps all your results in memory.

Comment (jeenbroekstra) [reply tools/federation/src/test/java/demos/MonitorRequestsDemo.java]: Yes. Typically not a big deal unless you do monstrously big queries - but you're of course correct that streaming for this kind of thing is more efficient. 

I'm fine with either way, also fine if you keep the example as-is @aschwarte10 - the point of this doc is not really to explore the different ways to process RDF4J query results of course :)

Comment (aschwarte10) [reply tools/federation/src/test/java/demos/MonitorRequestsDemo.java]: Thanks for the comments, very useful patterns for the examples, will keep them in mind for future changes

The purpose of this demo is is not the evaluation and consumption of the query, but rather the federation setup and specific configuration options. The system.out.println or in this example the count is just the simplest example. In here we in any case have something like 10 results only.

I would suggest to merge.

**Pull request merged**